core/src/com/unciv/logic/civilization/managers/TechManager.kt

```diff
@@ -155,6 +156,9 @@ class TechManager : IsPartOfGameInfoSerialization {
 
     fun isResearched(construction: INonPerpetualConstruction): Boolean = construction.requiredTechs().all{ requiredTech -> isResearched(requiredTech) }
 
+    /** resources which need no research count as researched */
+    fun isResearched(resource: TileResource) = resource.revealedBy?.let { isResearched(it) } ?: true
```

Comment:
This too seems strange, why are you involving this class in the logic of what constitutes a viewable resource? That logic already exists in other places 🤔

Reply:
Could you elaborate on that? If I understand you right, you agree with the method isResearched(resource) itself, but you question its usage in getExploredResourcesNotification(..)?

If the latter is the case, the answer is that I would like to address the problem as close as possible at its root: displaying the yellow notification of a resource which is not yet discovered. This should never succeed, no matter the caller of that method. Since that method is nullable already, it felt like the best place to do that check and not return any notification in case it shouldn't show one.

Reply:
Checking other uses of revealedBy I see that we are indeed missing a single place for this logic, might as well be here
I would change 2 things: 
- 2 liner with :?, no "let", to make the revealedBy==null case more explicit
- change function name to "isRevealed"